CONTRIBUTING.md

```diff
@@ -1,13 +1,115 @@
-Cocotb Contribution Guidelines
-==============================
+# Cocotb Contribution Guidelines
 
 Welcome to the cocotb development!
 We are an inclusive community with the common goal of improving the cocotb, a coroutine based cosimulation library for writing VHDL and Verilog testbenches in Python.
 This guide explains how to contribute to cocotb, and documents the processes we agreed on to manage the project.
 All processes in this document are designed to streamline the development effort, to avoid bottlenecks, and to ultimately give a pleasant experience to all involved.
 
-Architecture and Scope of Cocotb
---------------------------------
+
+## Setting Up a Development Environment
+
+Assuming you have used cocotb prior to reading this guide, you will already have the cocotb [installation prerequisites](https://docs.cocotb.org/en/latest/install.html) and standard development tools (editor, shell, git, etc.) installed.
+
+Additionally, you will need [doxygen](https://www.doxygen.nl/index.html), for building documentation, and [tox](https://pypi.org/project/tox/), for building documentation and running regression tests.
+
+We recommend if you are using a Linux distribution to use your system package manager to install doxygen.
+Likewise, doxygen can be installed using the homebrew package manager on Mac OS.
+Windows contributors should download a binary distribution installer from the main website.
+
+`tox` is a Python project and can be installed with `pip`.
+
+```command
+pip install tox
+```
+
+Finally, you should [fork and clone](https://guides.github.com/activities/forking/) the cocotb repo.
+This will allows you to make changes to cocotb source code, and run regressions and build documentation locally.
+
+Now you are ready to contribute!
+
+
+## Running Tests Locally
+
+First, [set up your development environment](#setting-up-a-development-environment).
+
+Our tests are managed by `tox`, which runs both `pytest` tests and our system of makefiles.
+The regression does not end on the first failure, but continues until all tests in the `/tests` and `/examples` directories have been run.
+
+To run the tests locally with `tox`, you will need to select an appropriate test environment.
+Valid test environments are formatted as `{your python version}-{your OS}`.
+Valid python version values are `py35`, `py36`, `py37`, `py38`, or `py39`;
```

Comment:
```suggestion
Valid python version values are e.g. `py36`, or `py39`;
```
Maybe formulate this as examples, not an exhaustive list, so that this is applicable in the future (``py310``) without changes.

Reply:
It would be nice to be able to point to what the list is though, which is the 3rd code section [here](https://tox.readthedocs.io/en/latest/example/basic.html#a-simple-tox-ini-default-environments). Put I cannot get a direct link to it.

Reply:
Those are the ones we list on the tox envlist as valid. It is intended to update that after the 1.5 release to remove 3.5 and add 3.10 to show which Python versions are supported.

Reply:
Then let's keep the full list and add a comment to the ``tox.ini``.